src/nvim/lua/executor.c

```diff
@@ -520,51 +621,18 @@ static int nlua_state_init(lua_State *const lstate) FUNC_ATTR_NONNULL_ALL
   lua_setfield(lstate, LUA_REGISTRYINDEX, "mpack.empty_dict");
   lua_setfield(lstate, -2, "_empty_dict_mt");
 
```

Comment:
we could move `vim.NIL` and `vim._empty_dict_mt` also into `nlua_common_vim_init`. just place the `nlua_empty_dict_ref` and `nlua_nil_ref` assigments within `if (!is_thread) { ... }` conditions.

Reply:
I've moved them and `vim.loop`.